### PR TITLE
Refactor/due date calculation

### PR DESCRIPTION
**Pull request by tamacodechi**
That was surprisingly a lot less work than I anticipated 😅 
(But also this is due to the fact that I split up the tasks instead of doing one huge PR)

### CODE REVIEW

File: lib/calculations/date_calculation.rb [hunk 1]
```diff
@@ -18,18 +18,7 @@ def invoice_date
...
-    return nil unless due_word.any?
-    due_date_label = Word.right_after(due_word.first)
-    date_after_label = if due_date_label.nil? || due_date_label.text != 'Date:'
-                         DateTerm.right_after(due_word.first)
-                       else
-                         DateTerm.right_after(due_date_label)
-                       end
-    date_after_label.to_datetime
+    return nil if DueDateLabelTerm.empty?
+    DateTerm.right_after(DueDateLabelTerm.first).to_datetime
```
Comment (clemenshelm) [inline lib/calculations/date_calculation.rb:22]: 100.000 times nicer 😄

File: spec/calculations/date_calculation_spec.rb [hunk 2]
```diff
@@ -145,17 +146,17 @@
       bottom: 1519
     )
 
+    DueDateLabelDetector.filter
     due_date_calculation = DateCalculation.new(
-      DateTerm.dataset
+      DueDateLabelTerm.dataset
```
Comment (clemenshelm) [inline spec/calculations/date_calculation_spec.rb]: Do we even need the argument here? In fact we access both `DateTerm` and `DueDateLabelTerm` directly in the calculation without using this argument.

**Review by clemenshelm — CHANGES_REQUESTED** [review]
Cool! But what I'm missing here: How is the due date label term found? It should have an own detector, shouldn't it?

File: spec/calculations/date_calculation_spec.rb (hunk 2 above)
Comment (tamacodechi) [reply spec/calculations/date_calculation_spec.rb]: We get:
``` 
ArgumentError:
       wrong number of arguments (given 0, expected 1) 
```

**Comment by tamacodechi** [comment]
It does have it's own detector :) The due date calculation just uses the terms found by the due date detector.

**Review by clemenshelm — APPROVED** [review]

**Pull request merged**